src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java

```diff
@@ -30,36 +28,36 @@
     // here merely to have a value to satisfy awaitTermination of the JAVA API
     private static final int MAX_EXECUTION_TIME_DAYS = 365;
 
-    public static final String ncksPath = "/usr/bin/ncks";
-    public static final String ncrcatPath = "/usr/bin/ncrcat";
-    public static final String ncdpqPath = "/usr/bin/ncpdq";
-
     private IndexReader indexReader = null;
     private final String profile;
     private final String subset;
-    private final Path outputFile;
+    private final Converter converter;
+    private Path outputFile;
     private final Integer limit;
     private Path baseTmpDir;
     private int threadCount = 1;
     private UserLog userLog = null;
     private ProgressListener progressListener = null;
+    private String mimeType = "application/x-netcdf";
+    private String extension = "nc";
 
-    protected GoGoDuck(String profile, String subset, String outputFile, Integer limit) {
+    protected GoGoDuck(String profile, String subset, String outputFile, Converter converter, Integer limit) {
```

Comment:
Should take an array of converters ideally and you should iterate on them. You can say we support just one now, but should still take an array. The necessary plumbing should probably be in place.